content/post/manage-mongodb-in-aks/index.md

```diff
@@ -0,0 +1,581 @@
+
+#### Export the Credentials
+
+KubeDB will create Secret and Service for the database `sample-mongodb-rs` that we have deployed. Let’s check them by following command,
+
+```bash
+$ kubectl get secret -n demo -l=app.kubernetes.io/instance=sample-mongodb-rs
+NAME                     TYPE     DATA   AGE
+sample-mongodb-rs-auth   Opaque   2      15m
+sample-mongodb-rs-key    Opaque   1      15m
+
+$ kubectl get service -n demo -l=app.kubernetes.io/instance=sample-mongodb-rs
+NAME                     TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)     AGE
+sample-mongodb-rs        ClusterIP   10.0.110.83   <none>        27017/TCP   15m
+sample-mongodb-rs-pods   ClusterIP   None          <none>        27017/TCP   15m
+```
+Now, we are going to use `sample-mongodb-rs-auth` to export credentials.
+Let’s export the `USER` and `PASSWORD` as environment variables to make further commands re-usable.
+
+```bash
+$ export USER=$(kubectl get secrets -n demo sample-mongodb-rs-auth -o jsonpath='{.data.\username}' | base64 -d)
+
+$ export PASSWORD=$(kubectl get secrets -n demo sample-mongodb-rs-auth -o jsonpath='{.data.\password}' | base64 -d)
+```
+
+#### Insert Sample Data
+
+In this section, we are going to login into our MongoDB database pod and insert some sample data. 
+
+```bash
+$ kubectl exec -it -n demo sample-mongodb-rs-0 -- mongo admin -u $USER -p $PASSWORD
+Defaulted container "mongodb" out of: mongodb, replication-mode-detector, copy-config (init)
+connecting to: mongodb://127.0.0.1:27017/admin?compressors=disabled&gssapiServiceName=mongodb
+
+
+rs:PRIMARY> show dbs
+admin   0.000GB
+config  0.000GB
+local   0.000GB
+
+rs:PRIMARY> show users
+{
+	"_id" : "admin.root",
+	"userId" : UUID("f01a3025-3937-4e87-b0d9-023f0994fc40"),
+	"user" : "root",
+	"db" : "admin",
+	"roles" : [
+		{
+			"role" : "root",
+			"db" : "admin"
+		}
+	],
+	"mechanisms" : [
+		"SCRAM-SHA-1",
+		"SCRAM-SHA-256"
+	]
+}
+
+rs:PRIMARY> use musicdb
+switched to db musicdb
+
+rs:PRIMARY> db.bands.insert({"name":"Backstreet Boys"});
+WriteResult({ "nInserted" : 1 })
+
+rs:PRIMARY> db.bands.find().pretty()
+{ "_id" : ObjectId("623aa7550d970b42f91471a3"), "name" : "Backstreet Boys" }
+
+rs:PRIMARY> exit
+bye
+
+```
+
+> This was just an example of our MongoDB ReplicaSet database deployment. The other databases that KubeDB supports are Elasticsearch, Redis, MySQL, PostgreSQL, MariaDB, Percona XtraDB, ProxySQL, Memcached and PgBouncer. More information about Run & Manage MongoDB on Kubernetes can be found [HERE](https://kubedb.com/kubernetes/databases/run-and-manage-mongodb-on-kubernetes/)
+
+## Backup MongoDB Database Using Stash
+
+Here, we are going to use Stash to backup the MongoDB database we deployed before.
+
+### Step 1: Install Stash
+
+Go to [Appscode License Server](https://license-issuer.appscode.com/) again to get the Stash Enterprise license.
```

Comment:
kubedb enterprise license works for stash too. So, no need to get Stash license.

Reply:
fixed